src/main/java/com/forgeessentials/economy/ModuleEconomy.java

```diff
@@ -177,25 +177,28 @@ public static void confirmNewWalletAmount(UserIdent ident, Wallet wallet)
 
     public static int tryRemoveItems(EntityPlayer player, ItemStack itemStack, int amount)
     {
-        int foundStacks = 0;
         int itemDamage = ItemUtil.getItemDamage(itemStack);
-        for (int slot = 0; slot < player.inventory.mainInventory.length; slot++)
-        {
-            ItemStack stack = player.inventory.mainInventory[slot];
-            if (stack != null && stack.getItem() == itemStack.getItem() && (itemDamage == -1 || stack.getItemDamage() == itemDamage))
-                foundStacks += stack.stackSize;
-        }
-        foundStacks = amount = Math.min(foundStacks, amount);
-        for (int slot = 0; slot < player.inventory.mainInventory.length; slot++)
+
+        for (int slot = 0; slot < player.inventory.mainInventory.length && amount > 0; slot++)
         {
             ItemStack stack = player.inventory.mainInventory[slot];
+
             if (stack != null && stack.getItem() == itemStack.getItem() && (itemDamage == -1 || stack.getItemDamage() == itemDamage))
             {
-                int removeCount = Math.min(stack.stackSize, foundStacks);
-                player.inventory.decrStackSize(slot, removeCount);
-                foundStacks -= removeCount;
+                int removeCount = Math.min(stack.stackSize, amount);
+
+                if (removeCount == stack.stackSize) {
+                    player.inventory.setInventorySlotContents(slot, null);
+                } else {
+                    player.inventory.decrStackSize(slot, removeCount);
+                }
+
+                amount -= removeCount;
             }
         }
+
+        player.inventoryContainer.detectAndSendChanges();
```

Comment:
Isn't this the only thing needed to sync with server side.  What is the reason for changing the rest of the logic?

Reply:
Please note that there were two identical loops that contained almost identical code with the only difference one loop counted the number of items, and the other deleted them. But it can be easily done in one loop.
Also I added `amount > 0` to loop exit condition on case the required number of items is removed.
Just small optimization.
(sorry for my english)

Reply:
The problem with optimizations is it requires additional testing as I don't know your code well enough to know if you broke existing functionality when adding your optimizations.  Also, please note that in the pull request you only stated that you fixed a client desync bug.  I have a policy of not accepting pulls that go beyond the stated scope.  Also, for a simple fix, (stated scope), how come you did not make a bug report?

Reply:
That is understandable.  However, making an issue is still good as it allows us to keep track of the bug.  I was unaware that this bug even existed.

Reply:
Yes, I understand. For some reason I didn't think at all about any organization. Of course it was a mistake.

Moreover, I just noticed I left a new error in the code: I forgot about the return value of `amount` variable. It no longer returns how many items have been removed from inventory. Please check it when you will reviewing my commit.

Reply:
Go ahead and make these fixes as an additional commit.  Make sure to test everything and verify that it is working.  Your logic looks sound for the other stuff though.